rclpy/rclpy/topic_endpoint_info.py

```diff
@@ -159,6 +159,12 @@ def __eq__(self, other):
 
     def __str__(self):
         gid = '.'.join(format(x, '02x') for x in self.endpoint_gid)
+        if self.qos_profile.history.value == QoSHistoryPolicy.UNKNOWN:
+            history_depth_str = self.qos_profile.history.name
+        else:
+            history_depth_str = ' '.join([
+                f'{self.qos_profile.history.name}', f'({self.qos_profile.depth})'
```

Comment:
nit: depth is only valid if the history kind is KEEP_LAST.
In KEEP_LAST case it's the depth field is ignored, so maybe we could not print it in that case.

another nit:

```python3
f'{self.qos_profile.history.name} ({self.qos_profile.depth})'
```

is way easier to read that the `' '.join([...])` thing.

Reply:
both makes sense, i will update them as you suggested.

Reply:
fixed in https://github.com/ros2/rclpy/pull/869